packages/fx-core/src/plugins/resource/aad/utils/aadManifestHelper.ts

```diff
@@ -152,4 +156,104 @@ export class AadManifestHelper {
 
     return result;
   }
+
+  public static validateManifest(manifest: AADManifest): string {
+    let warningMsg = "";
+
+    // if manifest doesn't contain name property or name is empty
+    if (!manifest.name || manifest.name === "") {
+      warningMsg += AadManifestErrorMessage.NameIsMissing;
+    }
+
+    // if manifest doesn't contain signInAudience property
+    if (!manifest.signInAudience) {
+      warningMsg += AadManifestErrorMessage.SignInAudienceIsMissing;
+    }
+
+    // if manifest doesn't contain requiredResourceAccess property
+    if (!manifest.requiredResourceAccess) {
+      warningMsg += AadManifestErrorMessage.RequiredResourceAccessIsMissing;
+    }
+
+    // if manifest doesn't contain oauth2Permissions or oauth2Permissions length is 0
+    if (!manifest.oauth2Permissions || manifest.oauth2Permissions.length === 0) {
+      warningMsg += AadManifestErrorMessage.Oauth2PermissionsIsMissing;
+    }
+
+    // if manifest doesn't contain preAuthorizedApplications
+    if (!manifest.preAuthorizedApplications) {
+      warningMsg += AadManifestErrorMessage.PreAuthorizedApplicationsIsMissing;
+    } else {
+      // if preAuthorizedApplications doesn't contain teams mobile/desktop client id
+      if (
+        !manifest.preAuthorizedApplications.find(
+          (item) => item.appId === TeamsClientId.MobileDesktop
+        )
+      ) {
+        warningMsg += AadManifestErrorMessage.TeamsMobileDesktopClientIdIsMissing;
+      }
+
+      // if preAuthorizedApplications doesn't contain teams web client id
+      if (!manifest.preAuthorizedApplications.find((item) => item.appId === TeamsClientId.Web)) {
+        warningMsg += AadManifestErrorMessage.TeamsWebClientIdIsMissing;
+      }
+    }
+
+    // if accessTokenAcceptedVersion in manifest is not 2
+    if (manifest.accessTokenAcceptedVersion !== 2) {
+      warningMsg += AadManifestErrorMessage.AccessTokenAcceptedVersionIs1;
+    }
+
+    // if manifest doesn't contain optionalClaims or access token doesn't contain idtyp clams
+    if (!manifest.optionalClaims) {
+      warningMsg += AadManifestErrorMessage.OptionalClaimsIsMissing;
+    } else if (!manifest.optionalClaims.accessToken.find((item) => item.name === "idtyp")) {
+      warningMsg += AadManifestErrorMessage.OptionalClaimsMissingIdtypClaim;
+    }
+
+    return warningMsg;
+  }
+
+  public static processRequiredResourceAccessInManifest(manifest: AADManifest): void {
+    const map = getPermissionMap();
+    manifest.requiredResourceAccess.forEach((requiredResourceAccessItem) => {
+      const resourceIdOrName = requiredResourceAccessItem.resourceAppId;
+      let resourceId = resourceIdOrName;
+      if (!isUUID(resourceIdOrName)) {
+        resourceId = map[resourceIdOrName].id;
```

Comment:
Will `map[resourceIdOrName]` be undefined?

Reply:
Update and add a test case